library/core/src/main/java/com/google/android/exoplayer2/util/Util.java

```diff
@@ -850,6 +855,17 @@ public static int inferContentType(String fileName) {
   }
 
   /**
+   * Returns the whether the device can output 32bit float audio
+   *
+   * @param capabilities The audio capabilities of the device.
+   * @param channelCount The channel count.
+   */
+  public static boolean canHandle32BitFloatAudio(AudioCapabilities capabilities, int channelCount) {
+    return (capabilities != null && capabilities.supportsEncoding(C.ENCODING_PCM_FLOAT)
+     && ((Util.SDK_INT > 20 && channelCount <= 2) || (Util.SDK_INT >= 24)));
```

Comment:
Any idea whether the change in support at API 24 was specific to NVIDIA Shield devices, or an Android platform limitation? I couldn't find any information about this internally. Have you found any documentation or changelogs that mention this, or is it based on observing the behavior on different API versions? Thanks.